src/main/java/io/lettuce/core/ConnectionState.java

```diff
@@ -113,36 +112,18 @@ protected void setUserNamePassword(List<char[]> args) {
         }
 
         if (args.size() > 1) {
-            setUsername(new String(args.get(0)));
-            setPassword(args.get(1));
+            this.credentialsSupplier = new DefaultCredentialsSupplier(new String(args.get(0)), args.get(1));
```

Comment:
In the case which _this.credentialsSupplier_ has a _delegate_ which takes the passwd from another service: after sending an _AUTH_ query to the server, we override the _credentialsSupplier_ that has _delegate_ with a new _DefaultCredentialsSupplier_ which  holds only the passwd at the time of the query.
In the case of a passwd rotation, the passwd at the time of the _AUTH_ query will be expired.
I think there is supposed to be a check for existence _credentialsSupplier_ and only if it does not exists, then to set it.

Reply:
Hi @mp911de, we're looking into adding support for recurring authentication using ephemeral passwords.
This will mean that the client will need to repeatedly reauthenticate, and on each reauthentication attempt receive a fresh password from the credentials supplier.
Overwriting the credentials supplier will mean that this flow will break after the first reauthentication.
I can see 2 option to come over it.

**The 1st:**
To check if the _this.credentialsSupplier_ is of type _DefaultCredentialsSupplier_, meaning there is no delegate by the customer.
```
if (!(credentialsSupplier instanceof DefaultCredentialsSupplier)) {
            return;
}
```
In order to be sure the customer didn't create a new _CredentialsSupplier_ which inherits from _DefaultCredentialsSupplier_, we put _DefaultCredentialsSupplier_ as final so no inheritance from it.

**The 2nd:**
To set _this.credentialsSupplier_ using the _DefaultCredentialsSupplier_ itself from the _RedisUri_ , and then to check if it has delegate or not, if there is delegate, meaning the user/passwd is supplied externally and not supposed be to be overridden by any auth command.

Anyway, I think we should change this function name as it doesn't always change the username/passwd anymore and also not on all cases

Reply:
When you add or remove a password on the Redis ACL, does that affect connected clients?

Generally as I understand the scenario, you want to call AUTH yourself so there doesn't seem to be need to have a credentials provider at all.